### PR TITLE
update nodejs version

### PR DESCRIPTION
**Pull request by Preeticp**
This PR updates the node.js version from 8 to 12 in the 3.11 version of docs.

### CODE REVIEW

**Comment by codyhoag** [comment]
@Preeticp out of curiosity, does this change relate to [ART-1806](https://issues.redhat.com/browse/ART-1806)? This is being discussed with the release team and how we should document it.

**Comment by Preeticp** [comment]
Hey @codyhoag, I was asked to work for this issue: https://issues.redhat.com/browse/APPSVC-264.
@akram and @waveywaves are the folks leading this and I see them mentioned on ART-1806
I am not sure if the 2 issues are the same, probably Akram and Vibhav can add more context.
I have currently asked the devs if the change needs to be made for 4.X version of the docs too. Awaiting their response.

**Comment by waveywaves** [comment]
@codyhoag This does relate to the aforementioned ART ticket.

**Comment by waveywaves** [comment]
LGTM

File: using_images/other_images/jenkins_slaves.adoc [hunk 1]
```diff
@@ -28,7 +28,7 @@ The first is a link:https://github.com/openshift/jenkins/tree/master/slave-base[
 Two more images that extend the base image are also provided:
 
 * link:https://github.com/openshift/jenkins/tree/master/agent-maven-3.5[Maven v3.5 image]
-* link:https://github.com/openshift/jenkins/tree/master/agent-nodejs-8[Node.js v8 image]
+* link:https://github.com/openshift/jenkins/tree/master/agent-nodejs-12[Node.js v12 image]
```
Comment (siamaksade) [inline using_images/other_images/jenkins_slaves.adoc:31]: Also "https://github.com/openshift/jenkins/tree/master/agent-nodejs-10[Node.js v10 image]" in addition to 12

File: using_images/other_images/jenkins_slaves.adoc [hunk 2]
```diff
@@ -58,7 +58,7 @@ $ docker pull registry.redhat.io/openshift3/jenkins-slave-base-rhel7
 $ docker pull registry.redhat.io/openshift3/jenkins-slave-maven-rhel7
 $ docker pull registry.redhat.io/openshift3/jenkins-slave-nodejs-rhel7
 $ docker pull registry.redhat.io/openshift3/jenkins-agent-maven-35-rhel7
-$ docker pull registry.redhat.io/openshift3/jenkins-agent-nodejs-8-rhel7
+$ docker pull registry.redhat.io/openshift3/jenkins-agent-nodejs-12-rhel7
```
Comment (siamaksade) [inline using_images/other_images/jenkins_slaves.adoc:61]: And `registry.redhat.io/openshift3/jenkins-agent-nodejs-10-rhel7`

File: using_images/other_images/jenkins_slaves.adoc [hunk 3]
```diff
@@ -74,7 +74,7 @@ $ docker pull registry.redhat.io/openshift3/jenkins-slave-base-rhel7
...
```
Comment (siamaksade) [inline using_images/other_images/jenkins_slaves.adoc:77]: And `registry.redhat.io/openshift3/jenkins-agent-nodejs-10-rhel7`

File: using_images/other_images/jenkins_slaves.adoc [hunk 4]
```diff
@@ -86,7 +86,7 @@ $ docker pull openshift/jenkins-slave-base-centos7
 $ docker pull openshift/jenkins-slave-maven-centos7
 $ docker pull openshift/jenkins-slave-nodejs-centos7
 $ docker pull openshift/jenkins-agent-maven-35-centos7
-$ docker pull openshift/jenkins-agent-nodejs-8-centos7
+$ docker pull openshift/jenkins-agent-nodejs-12-centos7
```
Comment (siamaksade) [inline using_images/other_images/jenkins_slaves.adoc:89]: And `openshift/jenkins-agent-nodejs-10-centos7`

Comment (Preeticp) [reply using_images/other_images/jenkins_slaves.adoc:89]: Thank you @siamaksade I will make these changes, I have no confirmation from @akram on changes in 4.X versions of the doc. Considering that .8 is EOL I think it makes sense to change it in 4.X too, WDYT?

Comment (siamaksade) [reply using_images/other_images/jenkins_slaves.adoc:92]: Agreed

**Comment by Preeticp** [comment]
@siamaksade all changes done, PTAL.

**Comment by akram** [comment]
@Preeticp , yes, agreed that we need to make changes also on the 4.x .
Will you make it in another PR?

Also, based on our yesterday's meeting, I have replied to @codyhoag with additions that we need to specify to allow users to switch the the new image version for their pipelines.

If all of these comes in another PR, I am OK with the current one and it LGTM

**Comment by Preeticp** [comment]
Thank you @akram!
More inline... 
> @Preeticp , yes, agreed that we need to make changes also on the 4.x .
Will you make it in another PR?

Yes, I will make these changes for 4.X versions through another PR. So let's get this one merged for 3.11.

> Also, based on our yesterday's meeting, I have replied to @codyhoag with additions that we need to specify to allow users to switch the the new image version for their pipelines.

Wrt the additions, probably @codyhoag would be the best person to answer.

**Comment by codyhoag** [comment]
@Preeticp I'm working on a release note that will be added for each minor version z-stream release. The notes will cover the manual process to upgrade from v8 &rarr; v10/v12. The release note will then reference the docs updated in this PR for more information on the images.

**Comment by Preeticp** [comment]
Thank you, Cody. I will go ahead and make these changes in another PR for 4.x

File: using_images/other_images/jenkins_slaves.adoc [hunk 5]
```diff
@@ -28,7 +28,7 @@ The first is a link:https://github.com/openshift/jenkins/tree/master/slave-base[
 Two more images that extend the base image are also provided:
 
 * link:https://github.com/openshift/jenkins/tree/master/agent-maven-3.5[Maven v3.5 image]
-* link:https://github.com/openshift/jenkins/tree/master/agent-nodejs-8[Node.js v8 image]
+* link:https://github.com/openshift/jenkins/tree/master/agent-nodejs-10[Node.js v10 image] and link:https://github.com/openshift/jenkins/tree/master/agent-nodejs-12[Node.js v12 image]
 
 The Maven and Node.js Jenkins agent images provide Dockerfiles for both CentOS
```
Comment (jboxman) [inline using_images/other_images/jenkins_slaves.adoc:33]: I don't know if we need a specific product attribute for _RHEL_ or _CentOS_.

**Review by jboxman — APPROVED** [review]
@Preeticp, I'm not sure about product attributes (and that's not actually directly in this PR anyway). Otherwise, LGTM.

**Comment by jboxman** [comment]
@Preeticp - we have these:

```
:op-system-first: Red Hat Enterprise Linux CoreOS (RHCOS)
:op-system: RHCOS
:op-system-base: RHEL
:op-system-base-full: Red Hat Enterprise Linux (RHEL)
```

File: using_images/other_images/jenkins_slaves.adoc (hunk 5 above)
Comment (jboxman) [reply using_images/other_images/jenkins_slaves.adoc:33]: Never mind, this is _enterprise-3.11_ so attributes aren't likely to apply here.

**Comment by Preeticp** [comment]
Hi @jboxman, if we are good with this PR, can you merge it, please?

**Comment by jboxman** [comment]
@Preeticp, done!

**Pull request merged**